src/wacks4square/square.py

```diff
@@ -0,0 +1,209 @@
+        logger.error(f"error in search orders request: {response}")
+        raise Exception(response.json())
+
+    def _get_order(self, order_id):
+        logger.info(f"making get order request for order id {order_id}")
+        response = requests.get(f"{BASE_API}/orders/{order_id}", headers=self.headers)
+        if response.ok:
+            data = response.json()
+            return data
+        logger.error(f"error getting order: {response}")
+        raise Exception(response.json())
+
+    def request_all_products(self):
+        """
+        A util function that is useful for figuring out the catalog IDs of werbies in
+        square
+
+        https://developer.squareup.com/reference/square/catalog-api/list-catalog
+        """
+        response = requests.get(
+            f"{BASE_API}/catalog/list", headers=self.headers, params={"types": ["ITEM"]}
+        )
+        products = response.json().get("objects")
+        all_variations = []
+        for product in products:
+            if product["is_deleted"]:
+                print("DELETED", product["item_data"]["name"])
+                continue
+            variation_str = product["item_data"]["name"]
+            for variation in product.get("item_data", {}).get("variations", []):
+                variation_str += (
+                    f'|{variation["item_variation_data"]["name"]}<{variation["id"]}>'
+                )
+            all_variations.append(variation_str)
+        print("\n\n".join(all_variations))
+
+    def _get_orders_since_timestamp(self, timestamp):
+        end_at = datetime.utcnow().isoformat()
+        params = {
+            "limit": 500,
+            "location_ids": LOCATION_IDS,
+            # return full order data, not just order id
+            "return_entries": False,
+            "query": {
+                "filter": {
+                    "date_time_filter": {
+                        "closed_at": {"start_at": timestamp, "end_at": end_at}
+                    },
+                    "state_filter": {"states": ["COMPLETED"]},
+                },
+                # there seems to be some implicit hard limits on the square API, so to get the real latest timestamp
+                # we need to sort by desc to ensure the most recent items aren't truncated
+                "sort": {"sort_field": "CLOSED_AT", "sort_order": "DESC"},
+            },
+        }
+        new_orders_response = self._search_orders(params)
+        return new_orders_response
+
+    def get_sales_since_timestamp(self, timestamp: Optional[str]) -> List[Sale]:
+        """
+        Query the square API for all orders since the given timestamp. Then transform these orders
+        into a list of Sale objects.
+
+        last_timestamp datetime string in isoformat, may be null if wacks is running for the first time
+        """
+        logger.info(f"getting num sales, timestamp {timestamp}")
+        if not timestamp:
+            timestamp = self._get_two_years_ago_in_isoformat()
```

Comment:
didn't werb say we should start after some date in december since she had a lot of test orders before that?